"""add question solution table

Revision ID: e855cd98d5e2
Revises: 7e82094aa100
Create Date: 2018-05-10 13:53:43.261757

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = 'e855cd98d5e2'
down_revision = '7e82094aa100'
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('questions',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('name', sa.String(length=100), nullable=False),
    sa.Column('pid', sa.Integer(), nullable=True),
    sa.Column('created_at', sa.Integer(), nullable=True),
    sa.Column('updated_at', sa.Integer(), nullable=True),
    sa.PrimaryKeyConstraint('id')
    )
    op.create_table('solutions',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('question_id', sa.Integer(), nullable=True),
    sa.Column('title', sa.String(length=100), nullable=False),
    sa.Column('content', sa.Text(), nullable=True),
    sa.Column('useful_count', sa.Integer(), nullable=True),
    sa.Column('useless_count', sa.Integer(), nullable=True),
    sa.Column('cause_describe', sa.Integer(), nullable=True),
    sa.Column('cause_product', sa.Integer(), nullable=True),
    sa.Column('cause_content_err', sa.Integer(), nullable=True),
    sa.Column('cause_operation', sa.Integer(), nullable=True),
    sa.Column('created_at', sa.Integer(), nullable=True),
    sa.Column('updated_at', sa.Integer(), nullable=True),
    sa.ForeignKeyConstraint(['question_id'], ['questions.id'], ),
    sa.PrimaryKeyConstraint('id')
    )
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_table('solutions')
    op.drop_table('questions')
    # ### end Alembic commands ###
